pkg/paho-mqtt/contrib/riot_iface.c

```diff
@@ -120,18 +121,20 @@ int NetworkConnect(Network *n, char *addr_ip, int port)
 {
     int ret =-1;
     sock_tcp_ep_t remote = SOCK_IPV4_EP_ANY;
-    char _local_ip[IP_MAX_LEN_ADDRESS];
 
-    strncpy(_local_ip, addr_ip, sizeof(_local_ip));
-    if (IS_USED(MODULE_IPV4_ADDR) &&
-        ipv4_addr_from_str((ipv4_addr_t *)&remote.addr, _local_ip)) {
+    ret = dns_query(addr_ip, &remote.addr, AF_UNSPEC);
+    if (ret > 0) {
+        remote.port = port;
+        remote.family = ret == 4 ? AF_INET : AF_INET6;
+    }
```

Comment:
Shouldn't this be wrapped into an if block:

```c
if (IS_USED(MODULE_SOCK_DNS)) {
...
}
```

or something like this.

Reply:
No need for that, if none of the DNS backends is compiled in, `dns_query()` will always just return `-ENOTSUP`.